src/pyfreeleh/row/gsheet.py

```diff
@@ -1,412 +1,458 @@
-        except Exception:
-            pass
+        If the pasesed in `columns` is empty, all columns will be returned.
 
-    def _ensure_headers(self) -> None:
-        self._wrapper.update_rows(self._spreadsheet_id, A1Range(self._sheet_name), [self._columns])
+        Args:
+            *columns: list of columns that we want to get.
 
-    def select(self, *columns: str) -> SelectStmt:
+        Returns:
+            SelectStmt: the select statement that is configured to return the selected columns.
+        """
         selected_columns = list(columns)
         if len(selected_columns) == 0:
-            selected_columns = self._columns[:-1]  # Exclude the _ts field.
+            selected_columns = self._columns
 
-        return SelectStmt(self._wrapper, self._spreadsheet_id, self._sheet_name, selected_columns, self._columns)
+        return SelectStmt(self._sheet_session, self._serializer, self._mapper, selected_columns)
 
-    def insert(self, rows: List[Dict[str, Any]]) -> InsertStmt:
-        # _ts field is required to help us select non-empty rows.
-        rows_with_ts = []
-        for row in rows:
-            row = row.copy()
-            row[self.TS_COLUMN_NAME] = time.time()
-            rows_with_ts.append(row)
+    def insert(self, rows: List[T]) -> InsertStmt[T]:
+        """Create the insert statement to insert given rows into the sheet.
 
-        return InsertStmt(self._wrapper, self._spreadsheet_id, self._sheet_name, self._columns, rows_with_ts)
+        Args:
+            rows: list of rows to be inserted.
 
-    def update(self, value: Dict[str, Any]) -> UpdateStmt:
-        return UpdateStmt(
-            self._wrapper, self._spreadsheet_id, self._sheet_name, self._scratchpad_cell, self._columns, value
-        )
+        Returns:
+            InsertStmt: the insert statement that is configured to insert the given rows.
+        """
+        return InsertStmt(self._sheet_session, self._serializer, rows)
 
-    def delete(self) -> DeleteStmt:
-        return DeleteStmt(self._wrapper, self._spreadsheet_id, self._sheet_name, self._scratchpad_cell, self._columns)
+    def update(self, update_value: Dict[str, Any]) -> UpdateStmt:
+        """Create the update statement to update rows on the sheet with the given value.
+
+        Args:
+            update_value: map of value by the field_name.
+
+        Returns:
+            UpdateStmt: the update statement that is configured to update the affected rows with the given value.
+
+        Examples:
+            To update all the rows (assume there's 10 rows) so that column `name` equals to `"cat"`:
 
-    def close(self) -> None:
-        self._ensure_initialised()
+            >> store.update({"name": "cat"}).execute()
+            10
+        """
+        update_value = update_value.copy()
+
+        pk_col_name = self._get_pk_field_name()
+        update_value.pop(pk_col_name, None)
```

Comment:
I think we shouldn't consider `_rid` as PK though. `_rid` is our own internal implementation and so I think it's okay to know that `_rid` exists.

Reply:
I pop this from the value so that they won't be able to change its value.